Libraries/LibGUI/TextEditor.cpp

```diff
@@ -1198,11 +1198,13 @@ void TextEditor::paste()
 {
     if (is_readonly())
         return;
-    auto paste_text = Clipboard::the().data();
-    printf("Paste: \"%s\"\n", paste_text.characters());
+    auto data_and_type = Clipboard::the().data_and_type();
+    if (data_and_type.type != "text")
+        return;
+    printf("Paste: \"%s\"\n", data_and_type.data.characters());
```

Comment:
Might be nice to use a [structured binding declaration](https://en.cppreference.com/w/cpp/language/structured_binding) here:

```cpp
auto [data, type] = clipboard.data_and_type();
if (type != "text")
    return;
printf("Paste: \"%s\"\n", data.characters());
...
```

Reply:
> feels like tuple unpacking in python

yes, I thought the same!